openedx/frontend-app-publisher
# fix: tiny mce button find error

## [PROD-3225](https://2u-internal.atlassian.net/browse/PROD-3255)

This PR fixes TinyMCE editor Language Selector error 
``` failed to find and set button ID in TinyMceLanguageSelectorPlugin after 3 attempts.```
The constant var `showMarketingFields` is used to show/hide the marketing fields in the course editor page but it was initialized with 
```
 const showMarketingFields = !currentFormValues.type || !courseTypes[currentFormValues.type]
      || courseTypes[currentFormValues.type].course_run_types.some((crt) => crt.is_marketable);
```
which is true if currentFormValues.type is null or undefined until CurrentFormValues is loaded from the server. So, the marketing fields were appearing until the CurrentFormValues is loaded from the server. This caused an issue where the TinyMCE language selector console-logged an error stating that it was unable to find the button. This occurred because the field was shown first and then removed based on the value of `showMarketingFields`, causing the TinyMCE language selector to encounter an error.

![image](https://github.com/openedx/frontend-app-publisher/assets/78806673/32af47fe-3fac-4784-88f8-b1b65eb2efd5)

------

### Testing Instructions:
- Create a Masters Only Course through publisher
- Go to the course editor page
- Verify that the marketing fields are not appearing
- Check from developer tools console that there is no error related to TinyMCE editor
